test/units/testsuite-19.delegate.sh

```diff
@@ -13,90 +13,94 @@ if [[ "$(get_cgroup_hierarchy)" != unified ]]; then
-useradd test ||:
-systemd-run --uid=test \
-            --property="User=test" \
-            --property="Delegate=yes" \
-            --slice workload.slice \
-            --unit test-workload0.scope\
-            --scope \
-            test -w /sys/fs/cgroup/workload.slice/test-workload0.scope -a \
-                 -w /sys/fs/cgroup/workload.slice/test-workload0.scope/cgroup.procs -a \
-                 -w /sys/fs/cgroup/workload.slice/test-workload0.scope/cgroup.subtree_control
+test_subgroup() {
+    unit="test-subgroup-$RANDOM.service"
+    systemd-run --wait \
+                --unit="$unit" \
+                --property="DynamicUser=1" \
+                --property="Delegate=pids" \
+                --property="DelegateSubgroup=foo" \
+                test -w "/sys/fs/cgroup/system.slice/$unit" -a \
+                     -w "/sys/fs/cgroup/system.slice/$unit/foo"
 
-# Verify that DelegateSubgroup= affects ownership correctly
-unit="test-subgroup-$RANDOM.service"
-systemd-run --wait \
-            --unit="$unit" \
-            --property="DynamicUser=1" \
-            --property="Delegate=pids" \
-            --property="DelegateSubgroup=foo" \
-            test -w "/sys/fs/cgroup/system.slice/$unit" -a \
-                 -w "/sys/fs/cgroup/system.slice/$unit/foo"
+    # Check that for the subgroup also attributes that aren't covered by
+    # regular (i.e. main cgroup) delegation ownership rules are delegated properly
+    if test -f /sys/fs/cgroup/cgroup.max.depth; then
+        unit="test-subgroup-$RANDOM.service"
+        systemd-run --wait \
+                    --unit="$unit" \
+                    --property="DynamicUser=1" \
+                    --property="Delegate=pids" \
+                    --property="DelegateSubgroup=zzz" \
+                    test -w "/sys/fs/cgroup/system.slice/$unit/zzz/cgroup.max.depth"
+    fi
 
-# Check that for the subgroup also attributes that aren't covered by
-# regular (i.e. main cgroup) delegation ownership rules are delegated properly
-if test -f /sys/fs/cgroup/cgroup.max.depth; then
+    # Check that the invoked process itsel is also in the subgroup
     unit="test-subgroup-$RANDOM.service"
     systemd-run --wait \
                 --unit="$unit" \
                 --property="DynamicUser=1" \
                 --property="Delegate=pids" \
-                --property="DelegateSubgroup=zzz" \
-                test -w "/sys/fs/cgroup/system.slice/$unit/zzz/cgroup.max.depth"
-fi
+                --property="DelegateSubgroup=bar" \
+                grep -q -x -F "0::/system.slice/$unit/bar" /proc/self/cgroup
+}
 
-# Check that the invoked process itsel is also in the subgroup
-unit="test-subgroup-$RANDOM.service"
-systemd-run --wait \
-            --unit="$unit" \
-            --property="DynamicUser=1" \
-            --property="Delegate=pids" \
-            --property="DelegateSubgroup=bar" \
-            grep -q -x -F "0::/system.slice/$unit/bar" /proc/self/cgroup
+test_controllers
```

Comment:
Please do this the new way:
1. Source `test-control.sh`;
2. Rename the test functions to `testcase_`;
3. Use `run_testcases` instead of calling the test functions one-by-one.